src/main.cpp

```diff
@@ -241,28 +241,61 @@ void receive_can() {
 }
 
 void handleTime() {
-	//TODO - change for D6
-	char thisGesture = 0;
-	int64_t currentTime = duration_cast<milliseconds>(clockResetTimer.elapsed_time()).count();
-
-	// update timeVal
-	if (g_lastGestureTime + DEBOUNCE_TIME < currentTime && g_lastGesture != thisGesture) {
-		if (currentTime <= g_lastGestureTime + GESTURE_MARGIN) {
-			currentTime = 0;
-			g_lastTime = -1;
-			g_lastGestureTime = 0;
-			clockResetTimer.reset();
-		} else {
-			g_lastGestureTime = currentTime;
-		}
-	}
-	
+
+	if(!buttonState[JOYSTICK_BUTTON]) {
+		clockResetTimer.reset();
+	} 
+
+	int64_t currentTime =clockResetTimer.read_ms();
+
 	timeVal.set(steering_time_t { (int)currentTime / 1000 / 60, (int)currentTime / 1000 % 60 });
-	g_lastGesture = thisGesture;
 }
 
 void runSteeringDisplay() {
 	while (1) {
 		display.run();
 	}
+}
+
+void updateShiftRegs() {
+	shiftLatch.write(1);
+    for(int i = 7; i >= 0; i--){
+		wait_us(500); // TODO: Debug hardware issue
```

Comment:
This can be removed now that we have solved the hardware issue

Reply:
Done